### PR TITLE
airbyte-ci: rework RC promotion

### PR DESCRIPTION
**Pull request by alafanechere**
## What
We want to use `-rc` suffix on release candidate.
It means we have to rework our promotion flow to cut release a main version  when the RC is deemed stable by the system.

Promoting a RC to a main version then just means: 
* Update its version to a non rc one
* Set `isReleaseCandidate` to false

## How
Rework the `airbyte-ci connectors publish --promote-release-candidate` command to:
* Bump the connector version to a main one
* Set `isReleaseCandidate` to false
* Commit and push all this to a PR with the `auto-merge` label.
* Update the changelog

Bonus: significant refacto on Step that modify files to have more shared logic and cleaner code.

## User Impact
- [x] Check that `/bump-version` still works following the refacto
- [x] Check that `up-to-date` still works following the refacto

### CODE REVIEW

**Comment by alafanechere** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/46675?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#46675** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/46675?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#46573** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/46573?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_manifest_only/manifest_component_transformer.py
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_manifest_only/manifest_component_transformer.py:1]: Unrelated change to make the linter happy

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_manifest_only/manifest_resolver.py
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_manifest_only/manifest_resolver.py:1]: Unrelated change to make the linter happy

**Review by clnoll — APPROVED** [review]
Approving so I don't block merge, but there are a few places where I'm afraid we'll swallow any errors that might happen, which will make it difficult to debug. If you could add some logging and possibly error handling too that would address my concern.

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py
Comment (clnoll) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py:675]: Okay cool - maybe add a comment explaining why we're okay continuing on failure?

**Pull request merged**
